.github/workflows/scripts/e2e-verify.common.sh

```diff
@@ -193,7 +193,7 @@ e2e_set_payload() {
     local this_builder
     this_builder=$(e2e_this_builder)
     if [[ "${this_builder}" == "gcb" ]]; then
-        jq -c ".provenance_summary.provenance[0].envelope.payload = \"$(echo "$2" | base64 -w0)\"" <"$1"
+        jq -c ".provenance_summary.provenance[0].envelope.payload = \"$(echo "$2" | base64 -w0)\"" <"$1" > tmp.json && mv tmp.json "$1"
```

Comment:
why do we need to create a file? I think the function needs to print the change, and the caller saves it to file of they want?

Reply:
ok. something isn't happening correctly here for gcb, then
https://github.com/slsa-framework/example-package/blob/441e65668f26e0018b31cc31955d6d8264304692/.github/workflows/scripts/e2e-verify.common.sh#L540-L546

Like `$bad_provenance` should actually be `$PROVENANCE`